pages/dao/[symbol]/proposal/components/instructions/Symmetry/AddTokenToBasketModal.tsx

```diff
@@ -0,0 +1,72 @@
+import Modal from "@components/Modal"
+import Input from "@components/inputs/Input"
+import { useEffect, useState } from "react"
+
+
+export const AddTokenToBasketModal = ({
+  open,
+  onClose,
+  supportedTokens,
+  onSelect
+}:{
+  open: boolean,
+  onClose: any,
+  supportedTokens: any,
+  onSelect: any
+}) => {
+  const [allTokens, setAllTokens] = useState(supportedTokens);
+  const [searchValue, setSearchValue] = useState('');
+
+  useEffect(() => {
+    if(searchValue.length > 0){
+      const filteredTokens = supportedTokens.filter((token: any) => {
+        return token.name.toLowerCase().includes(searchValue.toLowerCase()) || token.symbol.toLowerCase().includes(searchValue.toLowerCase())
+      })
+      setAllTokens(filteredTokens)
+    } else {
+      setAllTokens(supportedTokens)
+    }
+
+  }, [searchValue, supportedTokens])
+  return <>
+    {
+      open &&
```

Comment:
Do you need to have the `open` check here?
the modal itself take an `isOpen` variable.